indexing.sql

```diff
@@ -0,0 +1,67 @@
+\i scripts/car_models.sql;
+\i scripts/car_model_data.sql;
+\timing
+SELECT DISTINCT make_title
+FROM car_models
+WHERE make_code = 'LAM';
+\timing
+
+\timing
+SELECT DISTINCT model_title
+FROM car_models
+WHERE make_code = 'NISSAN' AND model_code = 'GT-R';
+\timing
+
+\timing
+SELECT make_code, model_code, model_title, year
+FROM car_models
+WHERE make_code = 'LAM';
+\timing
+
+\timing
+SELECT *
+FROM car_models
+WHERE year > 2009 AND year < 2016;
+\timing
+
+\timing
+SELECT *
+FROM car_models
+WHERE year = 2010;
+\timing
+
+-- CREATE INDEX idx_car_make ON car_models (make_code);
+-- CREATE INDEX idx_car_model ON car_models (model_code);
+-- CREATE INDEX idx_car_year ON car_models (year);
```

Comment:
So you should only make an index on make_code because your where clauses only really uses that one. You could also create a index on make_code and model_code since one of your where clauses uses it but it doesn't give you that much of a faster result. Year does not see the same benefits because for each year, there's still a lot of data, so even after sorting them by year, you'd still have to do a lot of searching.